src/modules/Dashboard/features/Participant/Assignments/AssignmentsTab/AssignmentsTab.hooks.test.tsx

```diff
@@ -0,0 +1,916 @@
+          }
+        },
+      );
+    });
+
+    test('should show Export modal', async () => {
+      const { rerender } = renderWithProviders(
+        <UseAssignmentsHookTest
+          activityOrFlow={mockParticipantActivities.autoAssignActivity}
+          targetSubject={mockedOwnerSubject}
+        />,
+        renderOptions,
+      );
+
+      userEvent.click(screen.getByTestId(`${testId}-activity-actions-dots`));
+      await waitFor(() => expect(screen.getByTestId(`${testId}-export`)).toBeVisible());
+
+      const exportDataButton = screen.getByTestId(`${testId}-export`);
+
+      // Attempt to export data without encryption key
+      fireEvent.click(exportDataButton);
+      expect(screen.queryByTestId(`${testId}-export-modal`)).toBeNull();
+      expect(screen.queryByTestId(`${testId}-export-modal-password`)).toBeVisible();
+
+      // Attempt to view data with encryption key
+      mockGetAppletPrivateKey.mockReturnValue('mockedPrivateKey');
+
+      rerender(
+        <UseAssignmentsHookTest
+          activityOrFlow={mockParticipantActivities.autoAssignActivity}
+          targetSubject={mockedOwnerSubject}
+        />,
+      );
+
+      expect(screen.queryByTestId(`${testId}-export-modal-password`)).toBeNull();
+      expect(screen.queryByTestId(`${testId}-export-modal`)).toBeVisible();
+    });
+  });
+
+  describe('Take Now', () => {
+    describe('should show or hide Take Now button depending on role', () => {
+      test.each`
+        canDoTakeNow | role                 | description
+        ${true}      | ${Roles.Manager}     | ${'Take Now for Manager'}
+        ${true}      | ${Roles.SuperAdmin}  | ${'Take Now for SuperAdmin'}
+        ${true}      | ${Roles.Owner}       | ${'Take Now for Owner'}
+        ${false}     | ${Roles.Coordinator} | ${'Take Now for Coordinator'}
+        ${false}     | ${Roles.Editor}      | ${'Take Now for Editor'}
+        ${false}     | ${Roles.Respondent}  | ${'Take Now for Respondent'}
+        ${false}     | ${Roles.Reviewer}    | ${'Take Now for Reviewer'}
+      `('$description', async ({ canDoTakeNow, role }: { canDoTakeNow: boolean; role: Roles }) => {
+        renderWithProviders(
+          <UseAssignmentsHookTest
+            activityOrFlow={mockParticipantActivities.autoAssignActivity}
+            targetSubject={mockedFullSubject1}
+          />,
+          {
+            ...renderOptions,
+            preloadedState: { ...renderOptions.preloadedState, ...getPreloadedState(role) },
+          },
+        );
+
+        const actionDots = screen.queryByTestId(`${testId}-activity-actions-dots`);
+        if (actionDots) userEvent.click(actionDots);
+
+        await waitFor(() => {
+          if (canDoTakeNow) {
+            expect(screen.getByTestId(`${testId}-activity-take-now`)).toBeVisible();
+          } else {
+            expect(screen.queryByTestId(`${testId}-activity-take-now`)).toBeNull();
+          }
+        });
```

Comment:
If you await this `userEvent.click`, you won't need the `waitFor` below it

```suggestion
        const actionDots = screen.getByTestId(`${testId}-activity-actions-dots`);
        await userEvent.click(actionDots);

        if (canDoTakeNow) {
          expect(screen.getByTestId(`${testId}-activity-take-now`)).toBeVisible();
        } else {
          expect(screen.queryByTestId(`${testId}-activity-take-now`)).toBeNull();
        }
```